File: Solutions/0643/0643.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2021/2/4 17:48
# File: 0643.py
# Desc: 
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        maxnum = sum(nums[:k])
        num = maxnum
        for i in range(1,len(nums)-k+1):
            num = num-nums[i-1]+nums[i+k-1]
            maxnum = max(maxnum, num)
        return maxnum/k

File: Solutions/0643/0643.java
/**
 * @Author: washing
 * @DateTime: 2021/02/04 17:48
 * @File: 0643.java
 * @Desc: 
 **/
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int biggest = 0;
        for (int i=0; i<k; i++){
            biggest += nums[i];
        }
        int num = biggest;
        for (int i=1; i<nums.length-k+1; i++){
            num = num - nums[i-1] + nums[i+k-1];
            biggest = num > biggest ? num : biggest;
        }
        return biggest/(double)k;
    }
}